### ISSUE TITLE
'Trainer' object has no attribute 'train_loader' error

### ISSUE THREAD

**Issue by fleventy-5** [Deci-AI/super-gradients#967]
I'm Trying to train yolo-nas on a custom dataset and I get the below error while running the training on google colab using 
```
trainer.train(model=model, 
              training_params=train_params, 
              train_loader=train_data, 
              valid_loader=val_data)
```
![image](https://github.com/Deci-AI/super-gradients/assets/78148524/b725c386-8b13-41d3-9644-6ceb16f7a69c)

**Comment by Louis-Dupont** [comment]
Hi @fleventy-5 
I think that the `train_loader` you passed is not a `Dataloader` but instead either `None`, `[]`, `False`, or any "0" value.
Can you please check it ?

**Comment by Louis-Dupont** [comment]
@momonoki3nenn, @fleventy-5 ,
I did not manage to reproduce it, but we pushed a change that should fix it. It will be in the next release, but meanwhile, you can install SG from our repo directly :)
``` shell
pip install git+https://github.com/Deci-AI/super-gradients
```

**Comment by Louis-Dupont** [comment]
@momonoki3nenn 
The `StopIteration` appears because we are trying to iterate over a Dataloader that is apparently empty (you can try `next(iter(train_data))` and you will get the same error)

Another thing that supports this hypothesis is the fact that on the right of the "Caching annotation" line, there is "1/1" written, which shows that your dataset doesn't include multiple images/labels.

So now the question is why is the dataloader empty ?
I would guess that either you dont point to the right path of your dataset or maybe your dataset doesnt have the right structrure. 
Feel free to check this [documentation page](https://docs.deci.ai/super-gradients/src/super_gradients/training/datasets/Dataset_Setup_Instructions.html#computer-vision-datasets-setup) to see how your data should be structured to use this (or another) dataset.
To test it simply, you can check the length or iterate over it `len(train_data)` or `next(iter(train_data))`.

**Comment by Vikram12301** [comment]
Is there any fix found for the _StopIteration_ issue?

**Comment by Louis-Dupont** [comment]
@Vikram12301 , did you install the nightly ?
``` shell
pip install git+https://github.com/Deci-AI/super-gradients
```
If yes, do you have a single sample, in your train set?
Please also provide the full snippet of code you are using to run, with among other dataset instantiation.

**Comment by Louis-Dupont** [comment]
@momonoki3nenn yeah it looks like. Did you try with `batch_size=1` ? My guess is that maybe you have `batch_size` > `len(dataset)`, which means that the dataloader cannot prepare any full batch.

Concerning your other error, it comes from the code of `YoloXDetectionLoss` but YoloNAS expects `PPYoloELoss` (like in the [notebooks](https://colab.research.google.com/drive/1q0RmeVRzLwRXW-h9dPFSOchwJkThUy6d#scrollTo=8TnXgisyefds&line=28&uniqifier=1))
If you are still working on YoloNAS, this could definitely explain your error. In that case, change like in the notebook to use `PPYoloELoss`.
If not, could you please share with me your code snippet with all of the code?

**Comment by Louis-Dupont** [comment]
@momonoki3nenn , can you try:
- `import sys; print(sys.getdefaultencoding())` to check what is your default encoding
- setting the environment variable `PYTHONIOENCODING` to `utf8`, and then run the training again

It looks like `tree` doesn't always encode to `utf8` on windows even though it is supposed to

**Comment by Louis-Dupont** [comment]
@momonoki3nenn, thanks for the investigation!
I don't understand why it fails in only very specific environments, so it's a bit hard for us to fix what we don't manage to reproduce ...

We might be able to fix it in SG if we understand exactly what leads to this encoding error (even if due to treelib bad implementation of encoding).

### First idea
The arrows might lead to this error. You can try to replace:
https://github.com/Deci-AI/super-gradients/blob/a30fa8fdc623533df785831f7457967066fb2ebe/src/super_gradients/training/utils/sg_trainer_utils.py#L41-L50
with
``` python
 def to_symbol(self) -> str: 
     """Get the symbol representing the current increase type""" 
     if self == IncreaseType.NONE: 
         return "" 
     elif self == IncreaseType.IS_GREATER: 
         return "[UP]" 
     elif self == IncreaseType.IS_SMALLER: 
         return "[DOWN]" 
     else: 
         return "=" 
```

### Second idea
The `colored()` function might lead to this error. You can try to replace:
https://github.com/Deci-AI/super-gradients/blob/a30fa8fdc623533df785831f7457967066fb2ebe/src/super_gradients/training/utils/sg_trainer_utils.py#L231-L237
``` python
diff_with_prev_colored = f"{monitored_value.has_increased_from_previous.to_symbol()} {change_from_previous}"
diff_with_best_colored = f"{monitored_value.has_increased_from_best.to_symbol()} {change_from_best}"
```

### Third idea
My third guess is to do both at the same time. 


If none of these works then it's probably just that the treelib doesnt even work with plain text in your case, which means that we need an alternative.
